### PR TITLE
More advanced auth

### PR DESCRIPTION
**Pull request by addaleax**
- try/catch reading + parsing the config file
- If config data is missing/invalid, prompt the user for data

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -1,10 +1,62 @@
...
+  // Ask the user for input and write to ~/.ncurc, then try again
+  process.stdout.write('Please enter your Github user information:\n' +
+      '[Github tokens can be created as described in ' +
+      'https://help.github.com/articles/' +
+      'creating-a-personal-access-token-for-the-command-line/]\n');
+  username = await prompt('Github user name');
+  token = await prompt('Github token');
+  check(username, token);
+  const json = JSON.stringify({ username, token }, null, '  ');
+  await writeFile(authFile, json);
```
Comment (jasnell) [inline lib/auth.js]: should set some reasonable default permissions on this file perhaps? owner read+write only perhaps?

Comment (addaleax) [reply lib/auth.js]: yes. :) done!

File: lib/request.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ async function request(query, variables) {
     uri: 'https://api.github.com/graphql',
     method: 'POST',
     headers: {
-      'Authorization': `Basic ${auth}`,
+      'Authorization': `Basic ${await auth()}`,
```
Comment (joyeecheung) [inline lib/request.js]: I think we should do this during start up instead of reading/prompting for crendentials everytime a request is sent?

Comment (addaleax) [reply lib/request.js]: @joyeecheung Ah, right, I *meant* to cache the value here :smile:

**Comment by joyeecheung** [comment]
Awesome work! Thanks!

**Pull request merged**